.github/workflows/publish.yaml

```diff
@@ -0,0 +1,80 @@
+name: Publish Juju Legend
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+       - master
+    paths:
+       - 'releases/**'
+
+jobs:
+  publish-images:
+    #if: github.event.pull_request.merged == true
+    name: Publish Charm images
+    runs-on: ubuntu-20.04
+    env:
+      CHARMCRAFT_AUTH: "${{ secrets.CHARMCRAFT_AUTH }}"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Install Docker
+        uses: docker-practice/actions-setup-docker@master
+
+      - name: Install Dependencies
+        run: |
+          sudo snap install jq
+          sudo snap install charmcraft --classic
+
+      - name: Upload Docker images to Charmhub
+        run: |
+          set -x
+
+          release_charm_with_image(){
+            charm_name="$1"
+            charm_resource="$2"
+            img_name="$3"
+            release_version="$4"
+
+            release_file="releases/${release_version}/manifest.json"
+
+            # Getting the release version.
+            img_version=$(cat "${release_file}" | jq --raw-output ".core.\"${img_name}\"")
+
+            # Pulling the image.
+            image="${img_name}:${img_version}"
+            docker pull "${image}"
+
+            # Get the image SHA.
+            image_sha=$(docker inspect --format '{{ .Id }}' ${image})
+
+            # Upload image to Charmhub. CHARMCRAFT_AUTH should be loaded in the environment, and
+            # it should have sufficient permissions to push the resource.
+            charmcraft upload-resource --quiet "${charm_name}" "${charm_resource}" --image=${image_sha}
+
+            # Get the last revision number and image revision number and release.
+            charm_rev=$(charmcraft revisions ${charm_name} | awk 'FNR == 2 {print $1}')
+            img_rev=$(charmcraft resource-revisions "${charm_name}" "${charm_resource}" | awk 'FNR == 2 {print $1}')
+            charmcraft release "${charm_name}" --revision=${charm_rev} --channel="${release_version}/edge" --resource="${charm_resource}:${img_rev}"
+
+            # Also release it as the latest.
+            charmcraft release "${charm_name}" --revision=${charm_rev} --channel=edge --resource="${charm_resource}:${img_rev}"
+          }
+
+          # Get the release files from the last commit.
+          for release in $(git diff --name-only --no-merges HEAD~1 | grep 'manifest.json')
```

Comment:
Wouldn't the "release" value returned from `git diff --name-only --no-merges HEAD~1 | grep 'manifest.json'` be something like:  release/2021-10-18/manifest.json

Instead of just the '2021-10-18' part which is what I think you're after?